### ISSUE TITLE
Samsung Lockscreen Denies My Password

### ISSUE THREAD

**Issue by ilovecookieee** [topjohnwu/Magisk#1794]
I patched boot.img and flashed it using odin. After successfully rooting my device, I can't login to my samsung account and the lockscreen denies my pattern/pin. After rebooting, the device bootloops. My device is SM-J810F Pie and I used Magisk Canary 19.4. Please. fix it. Thanks in advanced!

**Comment by ilovecookieee** [comment]
Sorry, but there's no TWRP available for my device that's why I just patched the AP/boot image and then flashed it using odin. 

I think I have the same issue with them:
https://github.com/topjohnwu/Magisk/issues/1696
https://github.com/topjohnwu/Magisk/issues/1617
https://github.com/topjohnwu/Magisk/issues/1682

Sorry I couldnt be more help.

**Comment by ilovecookieee** [comment]
Update:
I tried to root again and I followed the instruction in install.md. I start the adb logcat, set the pattern, locked my screen, and then tried to unlock it. Of course, the pattern is not working.

Here's the logcat:
[logcat.txt](https://github.com/topjohnwu/Magisk/files/3553848/logcat.txt)

There's an interesting line:
08-29 12:27:48.641  1153  2849 E LockSettingsStorage: Cannot read file java.io.FileNotFoundException: /data/system/fmmpassword.key: open failed: ENOENT (No such file or directory)

I think the pattern I set doesn't save that's why it's not working on lockscreen. Also, I still can't login to my samsung account. Always "Processing failed." Sorry for my bad english. Thank you in advance!

**Comment by dcoolArne** [comment]
Do you use Android 9? Because since Android 9 I need to use a custom kernel for my s9 and before I hade the exact same problem

**Comment by ilovecookieee** [comment]
Yeah I think there is a compatibility issue with magisk and my device's custom kernel. Sadly, there's no custom kernel for my device. I'll stick around with custom rom for now. Hopefully they can fix it in the near future. Yes, I'm on android pie btw

**Comment by Kopra159** [comment]
@topjohnwu 
@phhusson 

no way ??? 


bl destroy keystore 🥺


please give us guide or tell us what to do .??

today , I managed  to  crack lockscreen ..

there is something  will prevent open phone , deleting  lockscreen.db and .key  wasn't enough .

https://pastebin.com/KWLNRxqW

Here is logs :

![IMG_20191211_190904_759](https://user-images.githubusercontent.com/48631763/70643505-c6b28500-1c49-11ea-8e49-f3cf13c417fc.jpg)
![IMG_20191211_190902_046](https://user-images.githubusercontent.com/48631763/70643506-c74b1b80-1c49-11ea-8248-866f287c9e49.jpg)
![IMG_20191211_190911_312](https://user-images.githubusercontent.com/48631763/70643508-c74b1b80-1c49-11ea-9ccb-ce44fc2ffbdc.jpg)
![IMG_20191211_190907_573](https://user-images.githubusercontent.com/48631763/70643509-c7e3b200-1c49-11ea-9170-436e4552c531.jpg)
![IMG_20191211_190915_101](https://user-images.githubusercontent.com/48631763/70643510-c7e3b200-1c49-11ea-8d22-a212412eba8b.jpg)



please tell us what should we do ?

and thanks .

**Comment by Kopra159** [comment]
@topjohnwu 
@phhusson 


I know it isn't  magisk issue but please help .


This also failed 🥺🥺🥺: 

https://github.com/geckyn/android_device_samsung_j4primelte/commit/595912f4980fb4db974baf1b667941df4bc02745

+ 

create empty file in /etc .

**Comment by 488315** [comment]
You have to talk to your rom developer on xda. Not here

**Comment by Kopra159** [comment]
> You have to talk to your rom developer on xda. Not here

it stock rom .

custom lockscreen  was fixed .

Issue with stock rom .

rom dev is Samsung  😂😂😂😂😂

**Comment by Kopra159** [comment]
@ilovecookieee 
This solved issue with me but face unlock , doesn't  work , I will see it later .

chmod  0 /vendor/lib/hw/keystore.*.so

then reboot and set lockscreen  .

**Comment by phhusson** [comment]
Well this falls back to software AOSP keymaster instead of using
hardware keymaster, that's not as fix, more of a workaround.

**Comment by ilovecookieee** [comment]
@Kopra159 Nice. I already gave trying to root my phone but I'll try your workaround. Though I have a question, can you use the Galaxy store?

**Comment by Kopra159** [comment]
> @Kopra159 Nice. I already gave trying to root my phone but I'll try your workaround. Though I have a question, can you use the Galaxy store?

unfortunately  samsung acount doesn't  work after installing  twrp until now  .

**Comment by Kopra159** [comment]
> Well this falls back to software AOSP keymaster instead of using hardware keymaster, that's not as fix, more of a workaround. Le jeu. 12 déc. 2019 à 14:11, Mohamed Omar <notifications@github.com> a écrit :
> […](#)
> @ilovecookieee This solved issue with me but face unlock , doesn't work , I will see it later . chmod 0 /vendor/lib/hw/keystore.*.so then reboot and set lockscreen . — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Thanks .
Now ,  I understand  what happened  .

Replacing  hardware  keymaster  with aosp   keymaster  .
so  now lockscreen  works .


I am trying  since july 2018 until now  and matter happened  today  at the morning  😂😂.



Thanks and please if u got anyfix tell it to us .

**Comment by Kopra159** [comment]
> @Kopra159 Nice. I already gave trying to root my phone but I'll try your workaround. Though I have a question, can you use the Galaxy store?
sorry dude 

Samsung  account  works and galaxy store now .


![Screenshot_20191212-164330_Galaxy Store](https://user-images.githubusercontent.com/48631763/70721667-ac86ae80-1cfe-11ea-9a66-7bddf5f684ea.jpg)

**Comment by ilovecookieee** [comment]
Very nice! Thanks bro. So, I'll just run `chmod 0 /vendor/lib/hw/keystore.*.so`  then reboot right?

**Comment by Kopra159** [comment]
> Very nice! Thanks bro. So, I'll just run `chmod 0 /vendor/lib/hw/keystore.*.so` then reboot right?

yes , if u have twrp use it as it will be easier .

mount vendor 

then use terminal  and write command .



if u on system.
su 

# mount vendor as rw  : 
mount -o rw,remount /vendor
# if u are sar and  u faced problem with mount point 
mount -o rw,remount /

then command  .


don't  reboot until  u see before root  ---------  :- 
command  : 

ls -l /vendor/lib/hw/ |grep keystore
![Screenshot_20191212-170110_Terminal Emulator](https://user-images.githubusercontent.com/48631763/70723258-464f5b00-1d01-11ea-9542-c25eb5079d42.jpg)

**Comment by ilovecookieee** [comment]
Sadly, the workaround didn't work for me @Kopra159 .  I still can't login to my samsung account and the lockscreen is still broken. Btw, I use adb to change the permission after rooting and I don't have the `keystore.msm8937.so` in my `/vendor/lib/hw/`  
![image](https://user-images.githubusercontent.com/40349590/70777111-dfda4380-1db9-11ea-8a75-52f6895f8494.png)

**Comment by Kopra159** [comment]
> Sadly, the workaround didn't work for me @Kopra159 . I still can't login to my samsung account and the lockscreen is still broken. Btw, I use adb to change the permission after rooting and I don't have the `keystore.msm8937.so` in my `/vendor/lib/hw/`
> ![image](https://user-images.githubusercontent.com/40349590/70777111-dfda4380-1db9-11ea-8a75-52f6895f8494.png)

no it should .
u didn't  change  permission  in right  way   as it is 644 in picture not 0  .


Do u have twrp ???

**Comment by Kopra159** [comment]
@ilovecookieee 

Test this 
[Fix_Temp_lockscreen.zip](https://github.com/topjohnwu/Magisk/files/3961154/Fix_Temp_lockscreen.zip)


and send me result of check command

**Comment by Kopra159** [comment]
sorry guys call come when edit and my finger refer as issue  with mistake

**Comment by ilovecookieee** [comment]
Thanks for this [Fix_Temp_lockscreen.zip](https://github.com/topjohnwu/Magisk/files/3961154/Fix_Temp_lockscreen.zip) @Kopra159 . It flashed successfully. But after rebooting, it asked me for password. The phone is encrypted even if I didn't set any password.

**Comment by Kopra159** [comment]
> Thanks for this [Fix_Temp_lockscreen.zip](https://github.com/topjohnwu/Magisk/files/3961154/Fix_Temp_lockscreen.zip) @Kopra159 . It flashed successfully. But after rebooting, it asked me for password. The phone is encrypted even if I didn't set any password.

maybe u set password before that .

anyway 

easy flash this by twrp  : 

https://github.com/Kopra159/Prevent-Disabling-Lock-screen-app/releases/download/v2/crack.zip


then  flash  fix temp :

and give me result  of check command  .

**Comment by ilovecookieee** [comment]
> maybe u set password before that .  

Nope, I just flash a stock rom then root it. After changing the permission using `su -c 'chmod 0 /vendor/lib/hw/keystore.*.so'` in adb shell, I reboot my device. On start-up, it requires me to enter a password but I never set any password or pattern.

>give me result of check command  

Okay, I'll try it

**Comment by ilovecookieee** [comment]
@Kopra159 The `p=$(ls /data/system_de/0/spblob/ | grep -v "000000"  |awk '$0="/data/system_de/0/spblob/"$0')`  gave me an error: 

`ls: /data/system_de/0/spblob/: No such file or directory`

The fix temp was flashed successfully. After rebooting, the same problem on start up: *Your phone is encrypted for security. To start up your device, enter your password.* Again, I didn't add any password or pattern.

**Comment by Kopra159** [comment]
> @Kopra159 The `p=$(ls /data/system_de/0/spblob/ | grep -v "000000" |awk '$0="/data/system_de/0/spblob/"$0')` gave me an error:
> 
> `ls: /data/system_de/0/spblob/: No such file or directory`
> 
> The fix temp was flashed successfully. After rebooting, the same problem on start up: _Your phone is encrypted for security. To start up your device, enter your password._ Again, I didn't add any password or pattern.

ok , in twrp , try manually  


rm -f /data/system/locksettings.db
rm -f /data/system/*.key 



then reboot and see if password  is there ...
and tell me.

**Comment by Kopra159** [comment]
@ilovecookieee 

Failed  or no ? 🙄

============
Guys ,
biometric  won't  work  .

I solved the matter with face lock , using  opengapps trusted face and its libs .
![Screenshot_20191214-163735_Google Play services](https://user-images.githubusercontent.com/48631763/70854221-852c0f00-1ec1-11ea-907e-fb73f02aacd6.jpg)
![Screenshot_20191214-222935_Chrome](https://user-images.githubusercontent.com/48631763/70854223-89582c80-1ec1-11ea-942b-ba96784a3940.jpg)

**Comment by ilovecookieee** [comment]
@Kopra159  failed. It asked me again for password. i think I'm just gonna buy a new phone and it's not samsung lol. Anyway, thanks for all the help.

**Comment by gsoni411** [comment]
@Kopra159 
I'm on J810G After Rooting I'm having the same problem with the lock screen. I have a ss of the .keystore file located in my device , should I need to do any changes in tha so I can use my lock screen properly. 
I'm college student trying to learn root properly.
![Screenshot_20191215-230350_Solid Explorer](https://user-images.githubusercontent.com/58913923/70866416-71e77500-1f8f-11ea-8dec-3a4428ec4082.jpg)

**Comment by Kopra159** [comment]
> @Kopra159 failed. It asked me again for password. i think I'm just gonna buy a new phone and it's not samsung lol. Anyway, thanks for all the help.

ok dude ☹☹

**Comment by Kopra159** [comment]
> @Kopra159
> I'm on J810G After Rooting I'm having the same problem with the lock screen. I have a ss of the .keystore file located in my device , should I need to do any changes in tha so I can use my lock screen properly.
> I'm college student trying to learn root properly.
> ![Screenshot_20191215-230350_Solid Explorer](https://user-images.githubusercontent.com/58913923/70866416-71e77500-1f8f-11ea-8dec-3a4428ec4082.jpg)

give me files (u will find it in internal storage after excuting in terminal ):
su
ls -l /vendor/*/*  > /sdcard/vendor.txt
ls -l /system/*/*  > /sdcard/system.txt

**Comment by Kopra159** [comment]
@ilovecookieee 
@dcoolArne 
also use this app and send me 
screen shots .

https://play.google.com/store/apps/details?id=tk.hack5.treblecheck

**Comment by dcoolArne** [comment]
Here is a screenshot, but i use elementalx custom kernel with s9 stock rom
![Screenshot_20191216-003001_Treble Info](https://user-images.githubusercontent.com/39049038/70871073-7154db80-1f9b-11ea-987b-dfc1b0cc228b.jpg)

**Comment by gsoni411** [comment]
@Kopra159 
This is the file from my non rooted J810G

https://drive.google.com/file/d/1-QQXmNsVs8TGvSSJhNI6m0sigzlMZ-cB/view?usp=drivesdk

I've uprooted my phone using stock rom I'll send you the same file of my rooted in some time cuz I've a exam to appear today...

**Comment by ilovecookieee** [comment]
@Kopra159 I'm not rooted right now, but here's the screenshot. 
![Screenshot_20191216-132334_Treble Info](https://user-images.githubusercontent.com/40349590/70881426-7b142880-2007-11ea-9e1e-b1d34587587b.jpg)

**Comment by Kopra159** [comment]
> @Kopra159
> This is the file from my non rooted J810G
> 
> https://drive.google.com/file/d/1-QQXmNsVs8TGvSSJhNI6m0sigzlMZ-cB/view?usp=drivesdk
> 
> I've uprooted my phone using stock rom I'll send you the same file of my rooted in some time cuz I've a exam to appear today...

no need to it without  root .

I added su command  mean root .....dude

**Comment by Kopra159** [comment]
> @Kopra159 I'm not rooted right now, but here's the screenshot.
> ![Screenshot_20191216-132334_Treble Info](https://user-images.githubusercontent.com/40349590/70881426-7b142880-2007-11ea-9e1e-b1d34587587b.jpg)

ok no problem  here .


I need commands :


https://pastebin.com/jYtVJ58r


from root device  .
@ilovecookieee 
@gsoni411

**Comment by gsoni411** [comment]
This is the info From my J810G (without root)
i.e. on stock rom
![Screenshot_20191216-234902_Treble Info](https://user-images.githubusercontent.com/58913923/70933532-29e55280-2062-11ea-9f71-f64b20519883.jpg)

**Comment by kapmino269-MG** [comment]
@ilovecookieee 
@gsoni411 

try to flash that file :


https://androidfilehost.com/?fid=1395089523397964680


after rmm bypass v3 


then try lockscreen  🤔.


Maybe knox guard  apk and payment  service libs is the cause .


Try ....and tell me .

**Comment by tildebyte** [comment]
rats

Tried

```bash
su
chmod 0 /vendor/lib/hw/keystore.*.so
rm -f /data/system/locksettings.db
rm -f /data/system/*.key
```

as root on my Tab A 10.5.

The chmod *definitely* worked; I had to `mount -o rw,remount /vendor` and try again, then 'ls' showed the permissions changed to '------'

Likewise, 'ls' showed the locksettings.db and *.key files were gone.

Rebooted, and now I have a friendly

```text
Your tablet is encrypted for security.
To start up your device, enter your password.
```

Neither of those things are true, i.e. the device is **not** encrypted (and the reboot was waaay too fast for it to have applied encryption at that point), never has been, and no password has *ever* been set (I did the above shenanigans so I **could** (hopefully) set a password.

latest Magisk, su, hidden, running stock 9

Anyone?

**Comment by raspiduino** [comment]
I tried Magisk with stock kernel and it gave me problems like this :(

**Comment by rakeshbro** [comment]
same issue on j4+ stock kernel.
for now stock rom + root = no lockscreen 🙄

**Comment by hhihhio** [comment]
I used to successfully set the lock screen after root on samsung phone have this bug. I remember I've only install twrp go to recovery then reboot the phone prompt to wipe data, do not wipe data in twrp. Or some app ask you to set lock screen and here you go, don't set lock screen in phone settings
